DOCSP-30518: delete_many usage example (#58)
mongodb/docs-rust: source/usage-examples/delete-many.txt

.. _rust-usage-deletemany:

=========================
Delete Multiple Documents
=========================

You can delete multiple documents from a collection in a single operation
by calling the ``delete_many()`` method on a ``Collection`` instance.

To specify which documents to delete, pass a query filter that matches
the documents you want to delete. If you provide an empty document,
MongoDB matches all documents in the collection and deletes them. While
you can use ``delete_many()`` to delete all documents in a collection,
consider using the ``drop()`` method instead for better performance.

The ``delete_many()`` method returns a ``DeleteResult`` type, which contains
the ``deleted_count`` field. You can retrieve the total number of documents
deleted in your operation by printing the value of ``deleted_count``.

Example
-------

> This example accesses the ``planets`` collection in the ``sample_guides``
> database and deletes the documents representing the three closest planets
> to the sun. The following code filters for and deletes documents with an

Reviewer: Suggestion:
I think this can be made more concise and focused by describing the main functionality and omitting the description of referencing a database and collection:

```suggestion
This example deletes documents that represent the three closest planets
to the sun from the ``planets`` collection in the ``sample_guides``
database.
```
